crates/reth/node/src/gossip/package.rs

```diff
@@ -0,0 +1,359 @@
+
+        // Check we have enough bytes for seq_no (u64 = 8 bytes)
+        ensure!(
+            buf.remaining() >= SEQ_NO_SIZE,
+            "buffer too short for sequence number: need {SEQ_NO_SIZE} bytes, have {}",
+            buf.remaining()
+        );
+
+        // Decode the sequence number
+        let seq_no = buf.get_u64();
+
+        Ok(Self { header, seq_no })
+    }
+}
+
+impl AlpenGossipPackage {
+    /// Creates a new [`AlpenGossipPackage`].
+    pub(crate) fn new(message: AlpenGossipMessage, public_key: Buf32, signature: Buf64) -> Self {
+        Self {
+            message,
+            public_key,
+            signature,
+        }
+    }
+
+    /// Gets the message of the [`AlpenGossipPackage`].
+    pub fn message(&self) -> &AlpenGossipMessage {
+        &self.message
+    }
+
+    /// Gets the public key of the [`AlpenGossipPackage`].
+    pub fn public_key(&self) -> &Buf32 {
+        &self.public_key
+    }
+
+    /// Gets the signature of the [`AlpenGossipPackage`].
+    pub fn signature(&self) -> &Buf64 {
+        &self.signature
+    }
+
+    /// Validates the signature of the [`AlpenGossipPackage`].
+    pub fn validate_signature(&self) -> bool {
+        let message = self.message.encode().to_vec();
+        let hash = Buf32::from(eip191_hash_message(message).0);
+        let signature = self.signature();
+        let public_key = self.public_key();
+        verify_schnorr_sig(signature, &hash, public_key)
+    }
+
+    /// Encodes a [`AlpenGossipPackage`] into bytes.
+    pub(crate) fn encode(&self) -> BytesMut {
+        let mut buf = BytesMut::new();
+        let message = self.message.encode();
+        buf.put_slice(&message);
+        buf.put_slice(&self.public_key.0);
+        buf.put_slice(&self.signature.0);
+        buf
+    }
+
+    /// Decodes a [`AlpenGossipPackage`] from bytes with detailed error reporting.
+    pub(crate) fn try_decode(buf: &mut &[u8]) -> Result<Self> {
+        ensure!(!buf.is_empty(), "buffer is empty");
+
+        let message = AlpenGossipMessage::try_decode(buf)?;
+
+        // Check we have enough bytes for public key and signature
+        ensure!(
+            buf.remaining() >= Buf32::LEN + Buf64::LEN,
+            "buffer too short for public key and signature: need {} bytes, have {}",
+            Buf32::LEN + Buf64::LEN,
+            buf.remaining()
+        );
+
+        // Extract public key (32 bytes)
+        let mut public_key_bytes = [0u8; Buf32::LEN];
+        buf.copy_to_slice(&mut public_key_bytes);
+        let public_key = Buf32::from(public_key_bytes);
+
+        // Extract signature (64 bytes)
+        let mut signature_bytes = [0u8; Buf64::LEN];
+        buf.copy_to_slice(&mut signature_bytes);
+        let signature = Buf64::from(signature_bytes);
+
+        Ok(Self {
+            message,
+            public_key,
+            signature,
+        })
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Does it make sense to use `proptest`s?

Reply:
Yes, added `proptest!`s in 30303d52086ece5965cadc38fec7cb7ea8a839fa replacing the static ones.